scripts/wallet.js

```diff
@@ -1012,11 +1012,27 @@ export class Wallet {
         // Add primary output
         if (isDelegation) {
             if (!returnAddress) [returnAddress] = this.getNewAddress(1);
-            transactionBuilder.addColdStakeOutput({
-                address: returnAddress,
-                addressColdStake: address,
-                value,
-            });
+            // The per-output target for maximum staking efficiency
+            const nTarget = cChainParams.current.stakeSplitTarget;
+            // Keep track of the remainder, which is the 'unoptimal' output (or change/non-output, if too small)
+            let nRemainder = value;
+            // Generate optimal staking outputs
+            while (nRemainder > nTarget) {
```

Comment:
When `nRemainder <= COIN` the tx will contain only `value - nRemainder` PIVs instead of the requested `value`.

I suggest the following alternative approach, which also have the advantage that the behavior doesn't depend on `COIN`

Reply:
```suggestion
            for (let i == 0; i < Math.floor(value / nTarget); i++) {
                transactionBuilder.addColdStakeOutput({
                    address: returnAddress,
                    addressColdStake: address,
                    value: i == 0 ? nTarget + (value % nTarget) : nTarget,
                });
            }
```

basically the extra remainder in the range (0, `nTarget`) is added to the first tx output

Reply:
Good idea, removes the unnecessary `.addColdStakeOutput` block too, committing.